crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -140,28 +140,15 @@ def services
   # The method runs asynchronously, so there's a need to poll for the status and possible errors
   def nodes
     if params[:component]
-      case params[:component]
-      when "all"
-        ::Crowbar::UpgradeStatus.new.start_step(:nodes)
-        Api::Upgrade.nodes
-        head :ok
-      when "controllers"
-        render json: {
-          errors: {
-            nodes: {
-              data: "Not implemented yet."
-            }
-          }
-        }, status: :not_implemented
-      else
-        render json: {
-          errors: {
-            nodes: {
-              data: "Not implemented yet."
-            }
-          }
-        }, status: :not_implemented
-      end
+      # FIXME: we need to omit start of the step if:
+      # - it is running
+      # - controller substep is marked as finished (and there is no compute substep yet)
+      # - params[:component] is name of the node
+      # OR we could introduce new value for step status and mark the nodes step status as
+      # waiting/paused/* after the controllers are upgraded
```

Comment:
OR just do the start (for single node option )only when compute substep has failed